### PR TITLE
[fix](array)fix array_except/union for left const return only one row result

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes
if left param in array_except/union is const, it will make ret column only has one row column which will make result wrong

Issue Number: close #xxx

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/functions/array/function_array_set.h [hunk 1]
```diff
@@ -181,7 +181,11 @@ struct ArraySetImpl {
         constexpr auto execute_left_column_first = Impl::Action::execute_left_column_first;
         size_t current = 0;
         Impl impl;
-        for (size_t row = 0; row < left_data.offsets_ptr->size(); ++row) {
+        size_t row_size = left_data.offsets_ptr->size();
```
Comment (zhiqiang-hhhh) [inline be/src/vec/functions/array/function_array_set.h:184]: what if right const

Comment (amorynan) [reply be/src/vec/functions/array/function_array_set.h:184]: right and left is all not const ==>  left_data.offsets_ptr->size()
right const && left is not const ==> left_data.offsets_ptr->size()
right is not const && left is const ==> right.offsets_ptr->size()
right and left is all const ==>  left_data.offsets_ptr->size()

**Review by zhiqiang-hhhh — APPROVED** [review]

**Review by eldenmoon — APPROVED** [review]
lgtm

**Pull request merged**